### PR TITLE
Add documentation on controllers

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

**Comment by trasher** [comment]
https://glpi-developer-documentation--158.org.readthedocs.build/en/158/devapi/controllers.html

File: source/devapi/controllers.rst [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+The examples in this documentation use the magic ``__invoke`` method to force the controller to have only one action (see https://symfony.com/doc/current/controller/service.html#invokable-controllers).
+
+In general, this is recommended way to proceed but we do not force it and you are allowed to use multi actions controllers if you need them.
+
+Handling errors (missing rights, bad request, …)
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+A controller may throw some exceptions if it receive an invalid request.
+
+You can use any exception that extends Symfony\Component\HttpKernel\Exception, see below examples.
```
Comment (cedric-anne) [inline source/devapi/controllers.rst:166]:
```suggestion
You can use any exception that extends ``Symfony\Component\HttpKernel\Exception``, see below examples.
```

File: source/devapi/controllers.rst [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+   #[Symfony\Component\Routing\Attribute\Route("/Tickets", name: "glpi_tickets", methods: "GET")]
+
+Known limitation for ajax routes
+++++++++++++++++++++++++++++++++
+
+If an ajax route will be accessed by multiple POST requests without a page reload then you will run into CRSF issues.
+
+This is because GLPI’s solution for this is to check a special CRSF token that is valid for multiples requests, but this special token is only checked if your url start with ``/ajax``.
+
+You will thus need to prefix your route by /ajax until we find a better way to handle this.
```
Comment (cedric-anne) [inline source/devapi/controllers.rst:95]:
```suggestion
You will thus need to prefix your route by ``/ajax`` until we find a better way to handle this.
```

File: source/devapi/controllers.rst [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+Firewall
+^^^^^^^^
+
+By default, the GLPI firewall will not allow unauthenticated user to access your routes. You can change the firewall strategy with the ``Glpi\Security\Attribute\SecurityStrategy`` attribute.
+
+.. code-block:: php
+
+   <?php
+   #[Glpi\Security\Attribute\SecurityStrategy(Glpi\Http\Firewall::STRATEGY_NO_CHECK)]
+   public function __invoke(Symfony\Component\HttpFoundation\Reques $request): Response
```
Comment (cedric-anne) [inline source/devapi/controllers.rst:204]:
```suggestion
   public function __invoke(Symfony\Component\HttpFoundation\Request $request): Response
```

File: source/devapi/controllers.rst [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+   $response->headers->set('Content-Disposition', $disposition);
+   $response->headers->set('Content-Type', 'text/plain');
+   return $response
+
+Sending a file from disk
+++++++++++++++++++++++++
+
+.. code-block:: php
+
+   $file = 'path/to/file.txt';
```
Comment (cedric-anne) [inline source/devapi/controllers.rst:245]:
```suggestion
   <?php
   $file = 'path/to/file.txt';
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**